### ISSUE TITLE
Any plan for a Java client?

### ISSUE THREAD

**Issue by rsignell-usgs** [HDFGroup/h5serv#117]
Just wondering if there are plans for a Java high-level interface for `h5serv`, along the lines of the python high-level interface [`h5pyd`](https://github.com/HDFGroup/h5pyd)?

This would allow netcdf-java depedent clients like NASA's [Panoply](https://www.giss.nasa.gov/tools/panoply/) to more easily take advantage of object store on S3 by simply swapping a library, correct?   

Or does the netcdf-java library architecture make this simple library swap approach more challenging?

**Comment by jreadey** [comment]
No plans, but there is no technical reason why it couldn't be done.   Just as the h5pyd is a "Pure Python" package, using the HDF Rest API makes it much easier to create a pure Java library that reads and writes HDF5.

Panoply uses the NetCDF Java library, correct?  The easiest path might be to modify that library to utilize the REST API.

**Comment by rsignell-usgs** [comment]
Yes, Panoply, THREDDS Data Server and ERDDAP all use the NetCDF Java Library.    They may not like the fact that it would only work with NetCDF4, but they perhaps could develop a similar RESTful service for netcdf3 files.
